File: tiffany-frontend/src/redux/features/main/store/models/response/order-response.ts
/**
 * Order Response Types
 * Simplified type definitions matching backend OrderResponse
 */

import { OrderStatus } from "@/enums/order-status.enum";

export type PaymentMethod = "CASH" | "BANK";
export type PaymentStatus = "PAID" | "UNPAID" | "REFUNDED";

// Order Item Response - matches CartItemResponse structure for consistency
export interface OrderItemResponse {
  id: string;
  productId: string;
  productName: string;
  productImageUrl: string;
  productSizeId: string | null;
  sizeName: string;
  sku: string;
  quantity: number;

  // Display fields for UI rendering
  displayPrice: number;
  displayOriginPrice: number;
  displayPromotionType: string | null;
  displayPromotionValue: number | null;
  displayPromotionFromDate: string | null;
  displayPromotionToDate: string | null;
  hasActivePromotion: boolean;

  // Subtotal calculations
  subtotalBeforeDiscount: number;
  subtotalDiscountAmount: number;
  subtotalAfterDiscount: number;
}

export interface OrderResponse {
  id: string;
  createdAt: string;
  updatedAt: string;
  createdBy: string;
  updatedBy: string;
  orderNumber: string;
  customerId: string;
  customerName: string;
  customerPhone: string;
  customerEmail: string;
  customerNote: string | null;
  orderStatus: OrderStatus;
  paymentMethod: PaymentMethod;
  paymentStatus: PaymentStatus;
  subtotal: number;
  discountAmount: number;
  totalAmount: number;
  items: OrderItemResponse[];
}

File: tiffany-backend/src/main/java/com/tiffany/features/order/dto/response/OrderResponse.java
package com.tiffany.features.order.dto.response;

import com.tiffany.enums.order.OrderStatus;
import com.tiffany.enums.payment.PaymentMethod;
import com.tiffany.enums.payment.PaymentStatus;
import com.tiffany.shared.dto.BaseAuditResponse;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.math.BigDecimal;
import java.util.List;

@EqualsAndHashCode(callSuper = true)
@Data
public class OrderResponse extends BaseAuditResponse {
    private String orderNumber;
    private String customerId;
    private String customerName;
    private String customerPhone;
    private String customerEmail;
    private String customerNote;
    private OrderStatus orderStatus;
    private PaymentMethod paymentMethod;
    private PaymentStatus paymentStatus;

    private BigDecimal subtotal;
    private BigDecimal discountAmount;
    private BigDecimal totalAmount;

    private List<OrderItemResponse> items;
}
